src/server.ts

```diff
@@ -0,0 +1,62 @@
+import type { Server } from 'http';
+import { createServer as createHttpServer } from 'http';
+import { ApolloServer } from 'apollo-server-express';
+import type express from 'express';
+import type { Disposable } from 'graphql-ws';
+import { ApolloServerPluginDrainHttpServer } from 'apollo-server-core';
+import { PORT, isDevelopment } from './env';
+import { createContext, runSubscriptionServer } from './context';
+import { schemaWithMiddleware } from './schema';
+
+let serverCleanup: Disposable;
+
+const createApolloServer = (httpServer: Server): ApolloServer => new ApolloServer({
+  schema: schemaWithMiddleware,
+  context: createContext,
+  introspection: isDevelopment,
+  plugins: [
+    ApolloServerPluginDrainHttpServer({ httpServer }),
+    {
+      async serverWillStart() {
+        return {
+          async drainServer() {
+            serverCleanup?.dispose();
+          },
+        };
+      },
+    },
+  ],
+});
+
+const initializeApolloServer = (
+  httpServer: Server,
+  apollo: ApolloServer,
+  app: express.Application,
+  port: number,
+): (() => void) => {
+  apollo.applyMiddleware({ app });
+  serverCleanup = runSubscriptionServer(httpServer, apollo);
```

Comment:
Again, where is the subscription server requirement coming from? Why did you have to make any changes to the `createApolloServer`?